frontend/projects/upgrade/src/app/core/segments/store/segments.effects.ts

```diff
@@ -97,6 +101,24 @@ export class SegmentsEffects {
     )
   );
 
+  uspertFeatureFlagInclusionList$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(FeatureFlagsActions.actionAddFeatureFlagInclusionList),
+      map((action) => action.list),
+      withLatestFrom(this.store$.pipe(select(selectSelectedFeatureFlag))),
+      switchMap(([list, flag]) => {
+        const request = {
+          flagId: flag.id,
+          ...list,
+        };
+        return this.featureFlagDataService.addInclusionList(request).pipe(
+          map((listResponse) => FeatureFlagsActions.actionUpsertFeatureFlagInclusionListSuccess({ listResponse })),
+          catchError((error) => of(FeatureFlagsActions.actionUpsertFeatureFlagInclusionListFailure({ error })))
+        );
+      })
+    )
+  );
+
```

Comment:
Are we sure this code belongs in `segments.effects` instead of `feature-flags.effects?` It seems like all the `FeatureFlagsActions` are in that file.

Reply:
yeah i went back and forth, originally i had everything in feature flags, then you pointed out it was also a segments thing, so then it seemed more natural in the segments stuff, and then I ended up moving things mostly back over to feature flags... anyway I think you're right though I'll move it